monai/metrics/metric.py

```diff
@@ -15,121 +15,168 @@
+
+__all__ = ["Metric", "IterationMetric", "Cumulative", "CumulativeIterationMetric"]
 
 
 class Metric(ABC):
     """
-    Base class of all Metrics interface.
-    `__call__` is designed to execute metric computation.
+    Base class for metric computation for evaluating the performance of a model.
+    `__call__` is designed to execute the computation.
 
     """
 
     @abstractmethod
-    def __call__(self, *args: Any, **kwds: Any):
+    def __call__(self, *args: Any, **kwargs: Any):
         """
-        API to execute the metric computation.
-
+        This method should take raw model outputs as inputs, and return values that measure the models' quality.
         """
         raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
 
 
 class IterationMetric(Metric):
     """
-    Base class of Metrics interface for computation on a batch of tensors, usually the data of 1 iteration.
-    `__call__` is supposed to compute independent logic for several samples of `y_pred` and `y`(optional).
-    Usually, subclass only needs to implement the `_compute_tensor` function for computation process.
-    The input data shape should be `list of channel-first tensors` or a `batch-first tensor`.
+    Base class for metrics computation at the iteration level, that is, on a min-batch of samples
+    usually using the model outcome of one iteration.
+
+    `__call__` is designed to handle `y_pred` and `y` (optional) in torch tensors or a list/tuple of tensors.
 
+    Subclasses typically implement the `_compute_tensor` function for the actual tensor computation logic.
     """
 
     def __call__(self, y_pred: TensorOrList, y: Optional[TensorOrList] = None):  # type: ignore
         """
-        Execute basic computation for model prediction and ground truth.
-        It can support  both `list of channel-first Tensor` and `batch-first Tensor`.
-        And users can execute on every batch of data, then accumulate the results, or
-        accumulate the original `y_pred` and `y`, then execute on the accumulated data.
+        Execute basic computation for model prediction `y_pred` and ground truth `y` (optional).
+        It supports inputs of a list of "channel-first" Tensor and a "batch-first" Tensor.
 
         Args:
-            y_pred: the model prediction data to compute, must be a list of `channel-first` Tensor
+            y_pred: the raw model prediction data at one iteration, must be a list of `channel-first` Tensor
                 or a `batch-first` Tensor.
             y: the ground truth to compute, must be a list of `channel-first` Tensor
                 or a `batch-first` Tensor.
 
+        Returns:
+            The computed metric values at the iteration level.
+            The output shape could be a `batch-first` tensor or a list of `batch-first` tensors.
+            When it's a list of tensors, each item in the list can represent (batched) class-wise metrics.
```

Comment:
Could you please help also update this line to `a specific type of metric values.`?

Thanks.